src/token/erc6909/erc6909.cairo

```diff
@@ -0,0 +1,563 @@
+        /// @param spender The address of the spender.
+        /// @param approved The approval status.
+        fn setOperator(
+            ref self: ComponentState<TContractState>, spender: ContractAddress, approved: bool
+        ) -> bool {
+            ERC6909::set_operator(ref self, spender, approved)
+        }
+
+        /// @notice Checks if a contract implements an interface.
+        /// @param interfaceId The interface identifier, as specified in ERC-165.
+        /// @return True if the contract implements `interfaceId` and `interfaceId` is not 0xffffffff, false otherwise.
+        fn supportsInterface(self: @ComponentState<TContractState>, interface_id: felt252) -> bool {
+            ERC6909::supports_interface(self, interface_id)
+        }
+    }
+
+    /// internal
+    #[generate_trait]
+    pub impl InternalImpl<
+        TContractState, +HasComponent<TContractState>, impl Hooks: ERC6909HooksTrait<TContractState>
+    > of InternalTrait<TContractState> {
+        /// Creates a `value` amount of tokens and assigns them to `account`.
+        ///
+        /// Requirements:
+        ///
+        /// - `receiver` is not the zero address.
+        ///
+        /// Emits a `Transfer` event with `from` set to the zero address.
+        fn mint(
+            ref self: ComponentState<TContractState>,
+            receiver: ContractAddress,
+            id: u256,
+            amount: u256
+        ) {
+            assert(!receiver.is_zero(), Errors::MINT_TO_ZERO);
+            self.update(get_caller_address(), Zero::zero(), receiver, id, amount);
+        }
+
+        /// Destroys `amount` of tokens from `account`.
+        ///
+        /// Requirements:
+        ///
+        /// - `account` is not the zero address.
+        /// - `account` must have at least a balance of `amount`.
+        ///
+        /// Emits a `Transfer` event with `to` set to the zero address.
+        fn burn(
+            ref self: ComponentState<TContractState>,
+            account: ContractAddress,
+            id: u256,
+            amount: u256
+        ) {
+            assert(!account.is_zero(), Errors::BURN_FROM_ZERO);
+            self.update(get_caller_address(), account, Zero::zero(), id, amount);
+        }
+
+        /// Transfers an `amount` of tokens from `sender` to `receiver`, or alternatively mints (or burns) 
+        /// if `sender` (or `receiver`) is the zero address.
+        ///
+        /// This function can be extended using the `before_update` and `after_update` hooks. 
+        /// The implementation does not keep track of individual token supplies and this logic is left
+        /// to the extensions instead.
+        ///
+        /// Emits a `Transfer` event.
+        fn update(
+            ref self: ComponentState<TContractState>,
+            caller: ContractAddress, // For the `Transfer` event
+            sender: ContractAddress, // from
+            receiver: ContractAddress, // to
```

Comment:
+1
I think getting `caller` by calling `get_caller_address` in the function makes total sense and can reduce the amount of parameters

Reply:
Doing so would mean we're calling `get_caller_address` twice during the `transfer_from` function since the EIP defines a caller in the `Transfer` event. I agree it makes more sense to just remove the parameter from `update` and getting the caller in the function itself, but is it worth it considering we'd be making 2 `get_caller_address` calls in the same function? :thinking:

Reply:
Anyways, i changed the `update` function as discussed for now.